sklearn/base.py

```diff
@@ -433,7 +433,7 @@ def _validate_data(self, X, y='no_validation', reset=True,
                 X, y = check_X_y(X, y, **check_params)
             out = X, y
 
-        if check_params.get('ensure_2d', True):
+        if len(getattr(X, 'shape', ())) == 2:
             self._check_n_features(X, reset=reset)
```

Comment:
@thomasjpfan to make the new version of the `test_calibration_pipeline` test to pass while making it possible to check `n_features_in_` I had to make this small change in `BaseEstimator._validate_data`. I think this should be harmless but let me know if you think it is not.

Reply:
I like this proposal very much.